nhais-adaptor/edifact/models/tests/interchange_test.py

```diff
@@ -0,0 +1,83 @@
+        int_trl = InterchangeTrailer(number_of_messages=1, sequence_number="00001").to_edifact()
+        self.assertEqual(int_trl, "UNZ+1+00001'")
+
+
+class InterchangeTest(unittest.TestCase):
+    """
+    Test the generating of edifact message
+    """
+
+    def test_interchange_to_edifact(self):
+        expected_edifact_interchange = ("UNB+UNOA:2+SNDR+RECP+190423:0900+00001++FHSREG'"
+                                        "UNH+00001+FHSREG:0:1:FH:FHS001'"
+                                        "BGM+++507'"
+                                        "NAD+FHS+XX1:954'"
+                                        "DTM+137:201904230900:203'"
+                                        "RFF+950:G1'"
+                                        "S01+1'"
+                                        "RFF+TN:17'"
+                                        "NAD+GP+4826940,281:900'"
+                                        "HEA+ACD+A:ZZZ'"
+                                        "HEA+ATP+1:ZZZ'"
+                                        "DTM+956:20190423:102'"
+                                        "LOC+950+BURY'"
+                                        "S02+2'"
+                                        "PNA+PAT+N/10/10:OPI+++SU:STEVENS+FO:CHARLES+TI:MR+MI:ANTHONY+FS:JOHN'"
+                                        "DTM+329:20190420:102'"
+                                        "PDI+1'"
+                                        "NAD+PAT++MOORSIDE FARM:OLD LANE:ST PAULS CRAY:ORPINGTON:KENT+++++BR6 7EW'"
+                                        "UNT+18+00001'"
+                                        "UNZ+1+00001'")
+
+        date_time = "2019-04-23 09:00:04.159338"
+
+        msg_bgn = MessageBeginning(party_id="XX1", date_time=date_time, ref_number="G1")
+        msg_trg_1 = MessageSegmentTrigger1(transaction_number=17,
+                                           party_id="4826940,281",
+                                           acceptance_code="A",
+                                           acceptance_type="1",
+                                           date_time="2019-04-23 09:00:04.159338",
+                                           location="Bury")
+        patient_name = Name(family_name="Stevens", first_given_forename="Charles", title="Mr", middle_name="Anthony",
+                            third_given_forename="John")
+        patient_address = Address(house_name="MOORSIDE FARM", address_line_1="OLD LANE",
+                                  address_line_2="ST PAULS CRAY", town="ORPINGTON", county="KENT", post_code="BR6 7EW")
+
+        msg_trg_2 = MessageSegmentTrigger2(id_number="N/10/10", name=patient_name, date_of_birth="2019-04-20",
+                                           gender="1", address=patient_address)
+
+        msg = Message(sequence_number="00001", message_beginning=msg_bgn, message_segment_trigger_1=msg_trg_1,
+                      message_segment_trigger_2=msg_trg_2)
+        msgs = Messages([msg])

```

Comment:
Could these longer initialisations be formatted consistently? And I couldn't find variable naming conventions for python but I know in C# I think you attempt to avoid shortened variable names, does the same apply in Python?

Reply:
I am not sure on best practice here for python. 
I suppose I want to make it clear what the named arguments are and shorter naming convention might no achieve that but this does make creating an instance of a class ridiculously long.
Do you think something like this is still clear?
`Message(seq_num="00001", msg_bgn=msg_bgn, msg_sgt_tgr_1=msg_trg_1, msg_sgt_tgr_2=msg_trg_2)`

Reply:
Just to document this I was referring to the actual variable names (`msg`, `msgs`, `msg_trg_2`) but we decided it's acceptable convention to shorten variable names in Python.